### ISSUE TITLE
[fhir-client] allow client to invoke custom ops with no parameters

### ISSUE THREAD

**Issue by lmsurpre** [IBM/FHIR#3333]
**Is your feature request related to a problem? Please describe.**
some of the custom ops like `$healthcheck` and `$versions` can be invoked with no parameters

however, our client api forces the user to construct an empty Parameters object and pass that to 
```
client.invoke(op, params);
```

i *think* the client always does a POST (even for custom ops that can be invoked via GET) and that seems reasonable to me.
however, I believe the server requires a valid Parameters object in the request body in the case of a POST.

**Describe the solution you'd like**
At the very least, we should provide a convenience function for clients to call invoke without passing a Parameters object.

If we don't update the server to allow empty-body requests to POST against custom ops, the client could just construct an empty Parameters object under the covers and send that

**Describe alternatives you've considered**
Update the client to support invoking custom ops via GET

**Acceptance Criteria**

1.  
   GIVEN [a precondition]
    AND [another precondition]
   WHEN [test step]
    AND [test step]
   THEN [verification step]
    AND [verification step]

**Additional context**

**Comment by prb112** [comment]
Hey Lee, 

I thought we had two flavors. 

1 with a resource.... one without.... 

```
    /**
     * Invokes the 'invoke' FHIR REST API operation.
     * @param operationName name of the operation to be performed
     * @param headers an optional list of request headers to be added to the request
     * @return a FHIRResponse that contains the results of the 'invoke' operation
     * @throws Exception
     */
    FHIRResponse invoke(String operationName, FHIRParameters parameters, FHIRRequestHeader... headers) throws Exception;

    /**
     * Invokes the 'invoke' FHIR REST API operation.
     * @param operationName name of the operation to be performed
     * @param resource the FHIR resource used in context for the operation
     * @param headers an optional list of request headers to be added to the request
     * @return a FHIRResponse that contains the results of the 'invoke' operation
     * @throws Exception
     */
    FHIRResponse invoke(String operationName, Resource resource, FHIRRequestHeader... headers) throws Exception;
```

Does that meet the needs or is there a nuance that I'm not following? 

Thanks

Paul

**Comment by lmsurpre** [comment]
Cool, I had missed the version that takes FHIRParameters.
That one definitely does invoke the operation via GET.
We should consider updating the javadoc to make that more clear.
Also, we should fix the `@param` list while we're in there (current its missing `@param parameters)

In either case, I was kind of hoping for an option where we can invoke an operation with no parameters and no resource.
do to the overload, this is an error:
```
client.invoke("healthcheck", null);
```

I suppose its not so difficult to new up a new FHIRParameters object and pass that.
However, in the case of `$reindex`, that doesn't work over GET and so we actually need to construct a Parameters object:
```
client.invoke("reindex", Parameters.builder().build());
```

Basically I was helping a user with this and it didn't feel as simple as it should be.
This wasn't the first time I had this thought and so opened this low-priority enhancement request.

**Comment by lmsurpre** [comment]
On the flip side, we have a lot of overloads on `invoke()` already:
<img width="553" alt="image" src="https://user-images.githubusercontent.com/5478798/153696294-185c39ea-b1fd-4ac5-99f3-3eadd65cb7f0.png">

I'm not sure if its worth it to add an option with no `FHIRParameters` and no `Resource`, but that was the idea.
Would be 3 new methods:
```
FHIRResponse invoke(String operationName, FHIRRequestHeader... headers) throws Exception;

FHIRResponse invoke(String resourceType, String operationName, FHIRParameters parameters, FHIRRequestHeader... headers) throws Exception;

FHIRResponse invoke(String resourceType, String operationName, String resourceId, FHIRRequestHeader... headers) throws Exception;
```
